scripts/_autoloads/elephant.gd

```diff
@@ -0,0 +1,137 @@
+extends Node
+
+signal event_logged(p_event: String)
+
+## Elephant is the module that drinks and knows things.
+## This scene will check and retain pertinent system and platform info.
+## It will also help save and load stuff from disk. It might also be the logger
+## module, but that is a lot of things.
+@export var save_game_constants: RSaveGame
+var platform_name:= ""
+var is_debug_build:= false
+var save_time: Dictionary
+var config_file_path: String
+var logger_file_path: String
+
+func _ready() -> void:
+	assert(save_game_constants != null, "Save Game Constants is empty!!")
+	is_debug_build = OS.is_debug_build()
+	print("Is dev build: ", is_debug_build)
+	if is_debug_build == true:
+		start_logger()
+	# Returns the name of the host OS.
+	# Note: Custom builds of the engine may support additional platforms,
+	# such as consoles, yielding other return values.
+	platform_name = OS.get_name()
+	match platform_name:
+		"Windows":
+			print("Platform: Windows")
+		"macOS":
+			print("Platform: Mac OS")
+		"Linux", "FreeBSD", "NetBSD", "OpenBSD", "BSD":
+			print("Platform: Linux/BSD")
+		"Android":
+			print("Platform: Android")
+		"iOS":
+			print("Platform: iOS")
+		"Web":
+			print("Platform: Web")
+	save_config()
+	setup_game_save_dir()
+	log_event("All Elephant ready initialization is done!")
+
+func check_dir_exists(p_dir_path: String) -> bool:
+	return DirAccess.dir_exists_absolute(p_dir_path)
+
+func setup_logger_dir() -> int:
+	if check_dir_exists(save_game_constants.save_log_dir_path) == false:
+		var state: Error = DirAccess.make_dir_absolute(save_game_constants.save_log_dir_path)
+		if state != OK:
+			assert(state == OK, "There was a problem trying to make a directory!!")
```

Comment:
Do we want to assert on being unable to create log file, or is an error/warning acceptable here?

Reply:
Hmm, well this is assuming we are running a devbuild and we want the logger to start, so I would say yes an assert is appropriate here.